src/core/include/openvino/op/scatter_elements_update.hpp

```diff
@@ -44,5 +44,65 @@ class OPENVINO_API ScatterElementsUpdate : public Op {
     bool evaluate_scatter_element_update(const HostTensorVector& outputs, const HostTensorVector& inputs) const;
 };
 }  // namespace v3
+namespace v12 {
+class OPENVINO_API ScatterElementsUpdate : public Op {
+public:
+    OPENVINO_OP("ScatterElementsUpdate", "opset12", op::Op);
+
+    enum class Reduction { NONE, SUM, PROD, MIN, MAX, MEAN };
+
+    ScatterElementsUpdate() = default;
+    /// \brief Constructs a ScatterElementsUpdate node
+
+    /// \param data            Input data
+    /// \param indices         Data entry index that will be updated
+    /// \param updates         Update values
+    /// \param axis            Axis to scatter on
+    ScatterElementsUpdate(const Output<Node>& data,
+                          const Output<Node>& indices,
+                          const Output<Node>& updates,
+                          const Output<Node>& axis,
+                          const Reduction reduction = Reduction::NONE,
+                          const bool use_init_val = true);
+
+    void validate_and_infer_types() override;
+
+    bool visit_attributes(AttributeVisitor& visitor) override;
+
+    Reduction get_reduction() const {
+        return m_reduction;
+    }
+
+    void set_reduction(const Reduction reduction) {
+        m_reduction = reduction;
+    }
```

Comment:
I don't think that we need public `set` method , not good to change it on fly. How do you think?

Reply:
It can be required e.g for GPU as they create operator using default ctor and then set attributes if required.

Reply:
As Paweł said, the setters and getters are used to manipulate the attribute values after the op has been constructed with the default ctor. You can have a look at the whole attribute visitor machinery as well as `NodeFactory` class, this is where those methods are used (and possibly in many places in the plugins). Each op in each opset follows this approach so I don't think we can skip the setters just for the op which is the subject of this particular PR.